packages/lib/src/io-command.ts

```diff
@@ -939,5 +966,298 @@ export abstract class SelectingInputAPICommand<I, L> extends SelectingInputAPICo
+			// @ts-ignore
+			const bv = b[this.sortKeyName2].toLowerCase()
+			return av === bv ? 0 : av < bv ? -1 : 1
+		})
+	}
+
+	protected buildListTableOutput2(sortedList: L2[]): string {
+		const definitions: TableFieldDefinition<L2>[] = this.listTableFieldDefinitions2 ?? [
+			this.sortKeyName2,
+			this.primaryKeyName2,
+		]
+		let count = 0
+		definitions.unshift({
+			label: '#',
+			value: () => (++count).toString(),
+		})
+		return this.tableGenerator.buildTableFromList(sortedList, definitions)
+	}
+
+	protected async processNormally(
+			idOrIndex: ID | string | undefined,
+			idOrIndex2: string | undefined,
+			listFunction: ListCallback<L>,
+			listFunction2: NestedListCallback<ID, L2>,
+			getFunction: NestedGetCallback<ID, ID2, O>): Promise<void> {
+
+		try {
+			if (idOrIndex) {
+				const id: ID = await this.translateToId(idOrIndex, listFunction)
+				if (idOrIndex2) {
+					const id2: ID2 = await this.translateToId2(id, idOrIndex2, listFunction2)
+					const item = await getFunction(id, id2)
+					this.writeOutput(item)
+				} else {
+					const list = this.sort2(await listFunction2(id))
+					writeOutputPrivate(list, this.outputOptions, this.buildListTableOutput2.bind(this))
+				}
+
+			} else {
+				const list = this.sort(await listFunction())
+				writeOutputPrivate(list, this.outputOptions, this.buildListTableOutput.bind(this))
+			}
+		} catch (err) {
+			this.logger.error(`caught error ${err}`)
+			process.exit(1)
+		}
+	}
+
+	static flags = outputFlags
+}
+export interface NestedListingOutputAPICommandBase<ID, ID2, O, L, L2> extends Outputting<O>, Listing<L> {}
+applyMixins(NestedListingOutputAPICommandBase, [Outputable, Outputting, Listing], { mergeFunctions: true })
+
+/**
+ * Use this as your base when you need to be able to list or get a single
+ * resource from a list that is itself a property of another listable resource.
+ *
+ * This class is used for commands where the second identifier is a simple string but the first is not.
+ * If you need a more complex second identifier type, you can use NestedListingOutputAPICommandBase class. If
+ * both identifiers are strings you can use the NestedListingOutputAPICommand2  class.
+ */
+export abstract class NestedListingOutputAPICommand<ID, O, L, L2> extends NestedListingOutputAPICommandBase<ID, string, O, L, L2> {
+	protected translateToId2 = stringTranslateToId2
+	static flags = NestedListingOutputAPICommandBase.flags
+}
+
+/**
+ * Use this as your base when you need to be able to list or get a single
+ * resource from a list that is itself a property of another listable resource.
+ *
+ * This class is used for commands where both identifiers are simple strings.
+ * If you need a more complex identifier type, you can use either the
+ * NestedListingOutputAPICommand or NestedListingOutputAPICommandBase class.
+ */
+export abstract class NestedListingOutputAPICommand2<O, L, L2> extends NestedListingOutputAPICommand<string, O, L, L2> {
```

Comment:
I'm trying to figure out a good way to name these without using numbers. Some questions to ask:

1. Which of these two classes will be more common, if any.
    * If one is definitely going to be the most used one, that one should get the name `NestedListingOutputAPICommand`.
   * If neither is the "normal" one, the one that uses both strings probably should get the simplest name (`NestedListingOutputAPICommand`) because that seems more consistent with the other classes we've done to me.
2. If one is less common, should we consider just dropping it altogether specifying `string` and exporting `stringTranslateToId` so commands can add the `protected translateToId = stringTranslateToId` themselves. (I should have exported `stringTranslateToId` anyway.)

Reply:
It's 50:50 right now -- capabilities and device profiles. Capabilities are the only things that require the use of these base classes, because of their id/version identifiers, so I guess both IDs being strings might be more common, but I don't know if we will run into this sort of structure again or not. I elected to use the "2" suffix for the classes where both IDs were strings. Hypothetically there could also be a variant where the first ID is a string and the second is not, though it seems unlikely. I'll look into your suggestion 2